src/sagas/workspaces.ts

```diff
@@ -357,7 +357,7 @@ export function* evalCode(
   context.runtime.debuggerOn =
     (actionType === actionTypes.EVAL_EDITOR || actionType === actionTypes.DEBUG_RESUME) &&
     context.chapter > 2;
-  if (!context.runtime.debuggerOn) {
+  if (!context.runtime.debuggerOn && context.chapter > 2) {
```

Comment:
Can we make this more logical, this part has always been making my brain explode.

```javascript
const useDebugger = actionType !== (actionTypes.EVAL_EDITOR || actionType !== actionTypes.DEBUG_RESUME) && context.chapter > 2;

if (useDebugger) {
  inspectorUpdate(undefined);
}
```

Honestly though, I don't even think the above code segment makes any sense. Please comment on what this code segment is even intended to do.

Reply:
Okay, `debuggerOn` is used as a flag for slang.

So the correct code should look like 

```javascript
context.runtime.debuggerOn = actionType !== actionTypes.EVAL_EDITOR && actionType !== actionTypes.DEBUG_RESUME && context.chapter > 2;

if (context.runtime.debuggerOn) {
  // Resets the inspector to prepare for new output
  inspectorUpdate(undefined);
}
```

Reply:
This was done by the INSPECT team, so I'm not certain about how it works either. Having said that, this:

> Okay, `debuggerOn` is used as a flag for slang.
> 
> So the correct code should look like
> 
> ```js
> context.runtime.debuggerOn = (actionType === actionTypes.EVAL_EDITOR || actionType === actionTypes.DEBUG_RESUME) && context.chapter > 2;
> 
> if (context.runtime.debuggerOn) {
>   // Resets the inspector to prepare for new output
>   inspectorUpdate(undefined);
> }
> ```

doesn't seem to be the same boolean expression as the original code. `!(actionType === actionTypes.EVAL_EDITOR || actionType === actionTypes.DEBUG_RESUME) && context.chapter > 2` will call `inspectorUpdate(undefined)` in the original code, but not in the suggested amendment.

Reply:
My mistake. The clause should be 

```javascript
if (!context.runtime.debuggerOn)
``` 

just like the original code

Reply:
No, this won't work because it can trigger `inspectedUpdate(undefined)` even when `!(context.chapter > 2)`, and that crashes the entire webpage. That's what the change was originally for.

Reply:
Ok, can we clean this up then. The logic here is very confusing. Can we add comments to clarify?